### PR TITLE
walletrpc: Fix SignPsbt for NP2WKH inputs

### PR DESCRIPTION
**Pull request by guggero**
Fixes https://github.com/lightningnetwork/lnd/issues/6626.

### CODE REVIEW

File: lnwallet/btcwallet/psbt.go [hunk 1]
```diff
@@ -347,7 +340,6 @@ func signSegWitV0(in *psbt.PInput, tx *wire.MsgTx,
 	if err != nil {
 		return fmt.Errorf("error signing input %d: %v", idx, err)
 	}
-	in.FinalScriptSig = scriptSig
```
Comment (Roasbeef) [inline lnwallet/btcwallet/psbt.go:350]: Great commit message, this makes a lot of sense in retrospect.

File: lntest/itest/lnd_psbt_test.go [hunk 2]
```diff
@@ -1019,6 +1123,178 @@ func assertPsbtSpend(ctx context.Context, t *harnessTest,
...
+	fundOutputs := map[string]uint64{
+		mainAddrResp.Address: 999000,
+	}
+	spendAddrTypes := []lnrpc.AddressType{
+		lnrpc.AddressType_NESTED_PUBKEY_HASH,
+		lnrpc.AddressType_WITNESS_PUBKEY_HASH,
+		lnrpc.AddressType_TAPROOT_PUBKEY,
+	}
+
+	for _, addrType := range spendAddrTypes {
```
Comment (Roasbeef) [inline lntest/itest/lnd_psbt_test.go:1279]: Nice expansion of our test coverage here 👍

**Review by Roasbeef — APPROVED** [review]
LGTM 🐊

**Review by bhandras — APPROVED** [review]
LGTM, very nice and clean PR. Only reviewed from the code quality POV as I'm not very knowledgeable about psbt construction.

**Pull request merged**